sdk/core/azure-core/src/main/java/com/azure/core/models/CloudEvent.java

```diff
@@ -0,0 +1,470 @@
+        "source",
+        "type",
+        "datacontenttype",
+        "dataschema",
+        "subject",
+        "time",
+        "data"
+    ));
+
+    /*
+     * An identifier for the event. The combination of id and source must be
+     * unique for each distinct event.
+     */
+    @JsonProperty(value = "id", required = true)
+    private String id;
+
+    /*
+     * Identifies the context in which an event happened. The combination of id
+     * and source must be unique for each distinct event.
+     */
+    @JsonProperty(value = "source", required = true)
+    private String source;
+
+    /*
+     * Event data specific to the event type.
+     */
+    @JsonProperty(value = "data")
+    private Object data;
+
+    /*
+     * Event data specific to the event type, encoded as a base64 string.
+     */
+    @JsonProperty(value = "data_base64")
+    private String dataBase64;
+
+    /*
+     * Type of event related to the originating occurrence.
+     */
+    @JsonProperty(value = "type", required = true)
+    private String type;
+
+    /*
+     * The time (in UTC) the event was generated, in RFC3339 format.
+     */
+    @JsonProperty(value = "time")
+    private OffsetDateTime time;
+
+    /*
+     * The version of the CloudEvents specification which the event uses.
+     */
+    @JsonProperty(value = "specversion", required = true)
+    private String specVersion;
+
+    /*
+     * Identifies the schema that data adheres to.
+     */
+    @JsonProperty(value = "dataschema")
+    private String dataSchema;
+
+    /*
+     * Content type of data value.
+     */
+    @JsonProperty(value = "datacontenttype")
+    private String dataContentType;
+
+    /*
+     * This describes the subject of the event in the context of the event
+     * producer (identified by source).
+     */
+    @JsonProperty(value = "subject")
+    private String subject;
+
+    @JsonIgnore
+    private Map<String, Object> extensionAttributes;
+
+    /*
+     * Cache serialized data for getData()
+     */
+    @JsonIgnore
+    private BinaryData binaryData;
+
+
+    /**
+     *
+     * @param source Identifies the context in which an event happened. The combination of id and source must be unique
+     *               for each distinct event.
+     * @param type Type of event related to the originating occurrence.
+     * @param data A {@link BinaryData} that wraps the original data, which can be a String, byte[], or model class.
+     * @param format Set to {@link CloudEventDataFormat#BYTES} to serialize the data to base64 format, or
+     *               {@link CloudEventDataFormat#JSON} to serialize the data to JSON.
+     * @param dataContentType The content type of the data. It has no impact on how the data is serialized but tells
+     *                        the event subscriber how to use the data. Typically the value is of MIME types such as
+     *                        "application/json", "text/plain", "text/xml", "application/+avro", etc. It can be null.
```

Comment:
FWIW, the Avro spec calls out that `avro/binary` is the `Content-Type`. https://avro.apache.org/docs/current/spec.html#HTTP+as+Transport